.github/workflows/ci.yml

```diff
@@ -0,0 +1,136 @@
+
+      - name: Install clang format
+        run: brew install clang-format
+
+      - name: nvm install
+        shell: bash -l {0}
+        run: nvm install
+
+      - name: nvm use
+        shell: bash -l {0}
+        run: nvm use
+
+      - name: Install bundler
+        run: gem install bundler --version 2.1.4
+
+      - name: Set Bundler Config
+        run: bundle config path vendor/bundle
+
+      - name: Cache Bundler install Gems
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os}}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Cache Pods
+        uses: actions/cache@v2
+        with:
+          path: Example/Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
+
+      - name: NPM ci
+        run: npm ci
+
+      - name: Bundle Install
+        run: bundle install --jobs 4 --retry 3
+
+      - name: Pod Install
+        run: bundle exec pod install
+        working-directory: Example
+
+      - name: Run NPM build
+        run: npm run build
+
+      - name: Run danger
+        run: bundle exec danger
+        if: ${{ matrix.is_main }}
+
+      - name: Check pristine
+        run: ./scripts/check-pristine-state package-lock.json
+
+
+      - name: Run Swiftlint
+        run: Pods/SwiftLint/swiftlint lint --strict --config .swiftlint.yml
+        working-directory: Example
+
+      - name: Build and verify docs
+        if: ${{ steps.extract_git_tag.outputs.GIT_TAG == '' }}
+        run: ./scripts/build-docs-ci && rm -rf docs
+
+      - name: Build docs
+        if: ${{ steps.extract_git_tag.outputs.GIT_TAG != '' }}
+        run: ./scripts/build-docs-ci ${{ steps.extract_git_tag.outputs.GIT_TAG }}
+
+      - name: Build and Test
+        env:
+          FULL_TEST: TRUE
+          BUILD_SDK: iphonesimulator14.0
+          DESTINATION: "OS=14.0,platform=iOS Simulator,name=iPhone 8"
+        run: bundle exec rake ci
+        working-directory: Example
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        if: github.ref == 'refs/head/master'
+        with:
+          personal_token: ${{ secrets.IOS_DOCS_DEPLOY_TOKEN }}
+          publish_dir: ./docs
+          keep_files: true
+          external_repository: backpack/ios
+          publish_branch: master
+
+  trigger_docs:
+    name: Trigger docs Travis CI build
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Trigger docs Travis CI
+        env:
+          TRAVIS_API_KEY: ${{ secrets.TRAVIS_API_KEY }}
+        run: |
+          curl -s -X POST -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Travis-API-Version: 3' -H 'Authorization: token $TRAVIS_API_KEY' -d '{"request": {"branch":"master"}}' https://api.travis-ci.org/repo/Skyscanner%2Fbackpack-docs/requests
+
```

Comment:
Should we do this after everything else so that it doesn't happen if something is broken?

Reply:
I considered this but I think the only thing the docs site needs is the screemshots which aren't tested anyway 🤔

Reply:
That is very true